include/poac/core/resolver/resolve.hpp

```diff
@@ -36,6 +36,11 @@ struct Package {
   ///
   /// Sometimes, this is like `1.66.0` or like `>=1.64.0 and <2.0.0`.
   String version_rq;
+
+  bool
+  is_conan() const noexcept {
+    return name.find("conan::") == 0;
```

Comment:
Actually, this is  what I want to discuss with you. For simplicity, I just present this simplest form. 

IMO, Conan is another type of package manager, whereas a registry is just another endpoint of packages for a specific package manager. They are orthogonal: package types define how we resolve dependencies, how we fetch the package; registry defines where to fetch the package.

In order to make your  code work, we need add another property `type` to registries.

```
[dependencies]
fmt = { version = "9.1.0", registry = "conan" }
spdlog = { version = "1.10.0", registry = "conan" }
leveldb = { version = "1.23", registry = "conan" }

# These are unneeded.
#[registries]
#conan = { index = "...", type = "conan" }
```

But this form calls for more typing, coders are lazy, this is why we don't like cmake.

At lease in next few years, conan may be the main source of packages for poac, so can we provide better syntax for it?

In implementation aspects, registries for different remotes can be handled the same, but registries for different kinds are handled differently.

Reply:
We might have lots of remotes but we probably only have a few package types. So provide shortcut for every package type is feasible.

Reply:
We can introduce the `type` key and also can provide default registries so that users do not need to type them.

If users just write `registry = "conan"` and no registries, then Poac infers `conan = { index = "...", type = "conan" }`

Once conan is no longer used, then we can remove the default registry from our codebase and require users to write registries.

WDYT?

Reply:
Seems ok, I'll work on it in the weekend.

Still, I think we should allow "conan::fmt = 9.1.0" to make our users happy. Conan and vcpkg may  be the main source of our packages for a long time. In the foreseeable future, poac packages won't be too much. Allowing 'vcpkg::fmt = 9.1.0' and 'conan::fmt = 9.1.0' will make our toml shorter, especially we have tens of dependencies. 

Do simple things simple.

Reply:
Thank you. I will think about it in my spare time, but at first glance, keeping data structured would be better as an aspect of storing data inside RDBMS.